doomed_patient/doomed_patient/NPC.cpp

```diff
@@ -10,18 +10,107 @@ NPC::NPC(PatientGame* game, VectorXY startCoordinates, Texture * sprite)
 	int random = rand() % 4;
 	movementDirection = static_cast<Directions::Direction>(random);
 }
+
+
 void NPC::move(Directions::Direction direction)
 {
 	centre += Directions::getDirectionVector(direction) * speed;
 }
+
+
 void NPC::update(LevelCell::Passable blocked)
 {
+	// Makes sure NPC's grid position is up to date
+	updateCurrentCell();
+
+	// Update direction if necessary
+	updateDirection();
+
+	move(movementDirection);
+}
+
+// Call NPC behaviour code/methods that change NPC movementDirection from this method!
+void NPC::changeDirection()
+{
+	std::shared_ptr<CellEdge> currentEdge = currentCell->getEdge(movementDirection);
+
+
+	//PUT STUFF HERE
+
+
+	// Temporary for testing
+	// Changes direction to a random direction until it finds one that isn't a wall
+	while (currentEdge->isWall())
+	{
+		int random = rand() % 4;
+		movementDirection = static_cast<Directions::Direction>(random);
+		currentEdge = currentCell->getEdge(movementDirection);
+	}
+}
+
+
+void NPC::updateDirection()
+{
```

Comment:
This is the method that ensures the NPC is past the centre of its stored grid coordinates, by comparing its centre with the cell's centre, relative to the NPC's movement direction